Sage/carbon
# fix(action-popover): add preventScroll to the focus method responsible for focusing each menu item

fix #6485

### Proposed behaviour



Adds `preventScroll` which will prevent supported browsers from scrolling the document to bring the newly-focused element into view. Fixes an issue where the page would scroll to the top when the`ActionPopover` component was opened.

### Current behaviour



Currently, when the `ActionPopover` component is rendered in tabular components, the page may scroll to the top when the component is opened - due to an issue where the ref is partially not available, so a scroll to the top of the page occurs to compensate

### Checklist




#### QA


### Additional context



### Testing instructions



**Click or press enter on the action popover in the In FlatTable [with highlightable rows ](https://carbon.sage.com/iframe.html?viewMode=docs&id=action-popover--action-popover-component&args=#in-flattable-with-highlightable-rows) story on master - you should automatically scroll to the top of the page (incorrect behaviour)**

**Click or press enter on the action popover in the In FlatTable [with highlightable rows ] story in this PR - you should not automatically scroll to the top of the page (correct behaviour)**

**Remove the `preventScroll` on line `218` in `action-popover-item.component.tsx` and ensure the added `pw` test fails**
